test/boot_storage_test.rb

```diff
@@ -89,6 +89,18 @@
 
       expect(subject.mbr_disk).to eq "/dev/vda"
     end
+
+    it "skips cache if storage gets changed" do
+      subject.RootPartitionDevice = "/dev/sda1"
+      subject.BootPartitionDevice = "/dev/sda2"
+      subject.mbr_disk = "/dev/sda"
+
+      allow(Yast::Storage).to receive(:GetTargetChangeTime).and_return(Time.now.to_i)
+
+      subject.detect_disks
+
+      expect(subject.RootPartitionDevice).to eq "/dev/vda1"
+    end
```

Comment:
see xdescribe on top of file. This test is disabled for now. As its testing require solid set of device graph mocking, which is currently a bit out of scope of this PR.